src/storage/entry.rs

```diff
@@ -0,0 +1,201 @@
+use super::*;
+use crate::join::Join;
+use hibitset::BitSetAll;
+
+impl<'e, T, D> Storage<'e, T, D>
+where
+    T: Component,
+    D: DerefMut<Target = MaskedStorage<T>>,
+{
+    /// Returns an entry to the component associated to the entity.
+    ///
+    /// Behaves somewhat similarly to `std::collections::HashMap`'s entry api.
+    ///
+    /// ## Example
+    ///
+    /// ```rust
+    /// # extern crate specs;
+    /// # use specs::prelude::*;
+    /// # struct Comp {
+    /// #    field: u32
+    /// # }
+    /// # impl Component for Comp {
+    /// #    type Storage = DenseVecStorage<Self>;
+    /// # }
+    /// # fn main() {
+    /// # let mut world = World::new();
+    /// # world.register::<Comp>();
+    /// # let entity = world.create_entity().build();
+    /// # let mut storage = world.write_storage::<Comp>();
+    ///  if let Ok(entry) = storage.entry(entity) {
+    ///      entry.or_insert(Comp { field: 55 });
+    ///  }
+    /// # }
+    /// ```
+    pub fn entry<'a>(&'a mut self, e: Entity) -> Result<StorageEntry<'a, 'e, T, D>, WrongGeneration>
+    where
+        'e: 'a,
+    {
+        if self.entities.is_alive(e) {
+            unsafe {
+                let entries = self.entries();
+                let (_, mut value): (BitSetAll, _) = entries.open();
+                Ok(Entries::get(&mut value, e.id()))
+            }
+        } else {
+            let gen = self
+                .entities
+                .alloc
+                .generation(e.id())
+                .unwrap_or(Generation::one());
+            Err(WrongGeneration {
+                action: "attempting to get an entry to a storage",
+                actual_gen: gen,
+                entity: e,
+            })
+        }
+    }
+
+    /// Returns a `Join`-able structure that yields all indices, returning
+    /// `Entry` for all elements
+    pub fn entries<'a>(&'a mut self) -> Entries<'a, 'e, T, D> {
```

Comment:
Example code would be nice to illustrate that you should join this with another storage.

Reply:
Added